code/game/machinery/biogenerator.dm

```diff
@@ -400,6 +400,44 @@
 		return
 	return ..()
 
+/obj/machinery/biogenerator/conveyor_act(var/atom/movable/AM, var/obj/machinery/conveyor/CB)
+	if(istype(AM, /obj/item/weapon/reagent_containers/glass))
+		var/obj/item/weapon/reagent_containers/glass/G = AM
+		if(beaker|| panel_open || G.w_class > W_CLASS_SMALL)
+			return FALSE
+		else
+			G.forceMove(src)
+			beaker = G
+			updateUsrDialog()
+	else if(processing)
+		return FALSE
+	else if(istype(AM, /obj/item/weapon/storage/bag/plants))
+		var/i = 0
+		for(var/obj/item/weapon/reagent_containers/food/snacks/grown/G in contents)
```

Comment:
this segment is weird, why not just look at contents length?

also why return false here if it contains more than 20, why not just remove 20 and stop at 20 as your code below does?

Reply:
Because this checks the content inside of the biogenerator instead of the bag, and beakers count as content.

Reply:
Ah yes. You know the machine should probably have an is_full() proc that does this or something but it's not a big deal.